AndroidSDK/src/com/leanplum/LeanplumNotificationChannel.java

```diff
@@ -0,0 +1,596 @@
+    }
+  }
+
+  /**
+   * Configures default notification channel, which will be used when channel isn't specified on
+   * Android O.
+   *
+   * @param context The application context.
+   * @param channel Default channel details.
+   */
+  static void configureDefaultNotificationChannel(Context context, JSONObject channel) {
+    if (context == null || channel == null) {
+      return;
+    }
+    Map<String, Object> defaultChannel = JsonConverter.mapFromJson(channel);
+
+    if (defaultChannel != null) {
+      String id = (String) defaultChannel.get("defaultNotificationChannel");
+      storeDefaultNotificationChannel(context, id);
+    }
+  }
+
+  /**
+   * Configures notification groups.
+   *
+   * @param context The application context.
+   * @param groups Notification groups.
+   */
+  static void configureNotificationGroups(Context context, JSONArray groups) {
+    if (context == null || groups == null) {
+      return;
+    }
+    List<HashMap<String, Object>> definedGroups = retrieveNotificationGroups(context);
+    List<HashMap<String, Object>> notificationGroups = JsonConverter.listFromJson(groups);
+
+    if (definedGroups != null) {
+      definedGroups.removeAll(notificationGroups);
+
+      // Delete groups that are no longer present.
+      for (HashMap<String, Object> group : definedGroups) {
+        if (group == null) {
+          continue;
+        }
+        String id = (String) group.get("id");
+        deleteNotificationGroup(context, id);
+      }
+    }
+
+    // Store newly received groups.
+    storeNotificationGroups(context, notificationGroups);
+
+    // Configure groups.
+    if (notificationGroups != null) {
+      for (HashMap<String, Object> group : notificationGroups) {
+        if (group == null) {
+          continue;
+        }
+        createNotificationGroup(context, group);
+      }
+    }
+  }
+
+  /**
+   * Retrieves stored notification channels.
+   *
+   * @param context The application context.
+   * @return List of stored channels or null.
+   */
+  private static List<HashMap<String, Object>> retrieveNotificationChannels(Context context) {
+    if (context == null) {
+      return null;
+    }
+    try {
+      SharedPreferences preferences = context.getSharedPreferences(Constants.Defaults.LEANPLUM,
+          Context.MODE_PRIVATE);
+      String jsonChannels = preferences.getString(Constants.Defaults.NOTIFICATION_CHANNELS_KEY,
+          null);
+      if (jsonChannels == null) {
+        return null;
+      }
+      JSONArray json = new JSONArray(jsonChannels);
+      return JsonConverter.listFromJson(json);
+    } catch (Exception e) {
+      Log.e("Failed to convert notification channels json.");
+    }
+    return null;
+  }
+
+  /**
+   * Retrieves stored default notification channel id.
+   *
+   * @param context The Application context.
+   * @return Id of default channel.
+   */
+  private static String retrieveDefaultNotificationChannel(Context context) {
+    if (context == null) {
```

Comment:
maybe add defaultNotificationChannelId and if it's null - get it from shared preferences? plus update it in storeDefaultNotificationChannel?